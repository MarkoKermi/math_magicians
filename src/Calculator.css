* {
  box-sizing: border-box;

  /* -webkit-user-select: none; */
  -webkit-user-drag: none;
}

body {
  background-color: burlywood;
  display: grid;
  height: 100vh;
  font-size: 2rem;
}

.container {
  height: 80vh;
  width: 50vh;
  background-color: black;
  border-radius: 1rem;
  padding: 1rem;
  -webkit-box-shadow: 7px 7px 5px 0 rgba(50, 50, 50, 0.75);
  -moz-box-shadow: 7px 7px 5px 0 rgba(50, 50, 50, 0.75);
  box-shadow: 7px 7px 5px 0 rgba(50, 50, 50, 0.75);
  margin: auto;
  margin-right: 10%;
}

.wrapper {
  display: grid;
  height: 100%;
  width: 100%;
  grid-template-columns: repeat(4, 1fr);
  gap: 10px;
}

.screen {
  grid-column: 1/ -1;
  text-align: end;
  padding: 0.5rem;
  font-size: 4rem;
  overflow: hidden;
}

.btn {
  display: grid;
  place-content: center;
  background-color: rgb(54, 52, 52);
  border-radius: 50%;
  color: white;
  cursor: pointer;
  font-weight: bold;
  font-size: x-large;
}

.btn:hover {
  background-color: gray;
}

.light-gray {
  background-color: gray !important;
  color: black !important;
}

.light-gray:hover {
  background-color: white !important;
}

.orange {
  background-color: orange !important;
}

.orange:hover {
  background-color: white !important;
  color: orange !important;
}

.zero {
  grid-column: 1/ span 2;
  border-radius: 25px;
  place-content: unset;
  align-items: center;
  padding-left: 2rem;
}
