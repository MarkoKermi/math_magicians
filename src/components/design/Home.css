.homeContainer {
  padding: 0 30px;
}

.homeParagraphs {
  padding: 0 30px;
}

p {
  margin: 5%;
}
