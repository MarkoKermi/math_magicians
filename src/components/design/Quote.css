.quoteDiv {
  text-align: center;
  display: flex;
  justify-content: center;
  align-items: center;
  height: calc(100vh - 120px);
}

.quote {
  width: 80%;
}

h2 {
  display: flex;
  flex-direction: column;
  margin: 10%;
}
