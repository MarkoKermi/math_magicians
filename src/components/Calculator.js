import React, { useState } from "react";
import "./Calculator.css";
// import calculate from "./logic/calculate";
import calculate from "./logic/calculate";

function Calculator() {
  const [result, setResult] = useState({
    total: null,
    next: null,
    operation: null,
  });

  const handleClick = (buttonName) => {
    setResult(calculate(result, buttonName));
  };

  return (
    <div className="container">
      <div className="wrapper">
        <div
          className="screen"
          type="text"
          value={result.next || result.total || "0"}
        />
        <div className="btn light-gray" onClick={() => handleClick("AC")}>
          AC
        </div>
        <div className="btn light-gray" onClick={() => handleClick("%")}>
          %
        </div>
        <div className="btn light-gray" onClick={() => handleClick("+/-")}>
          +/-
        </div>
        <div className="btn orange" onClick={() => handleClick("/")}>
          /
        </div>
        <div className="btn" onClick={() => handleClick("7")}>
          7
        </div>
        <div className="btn" onClick={() => handleClick("8")}>
          8
        </div>
        <div className="btn" onClick={() => handleClick("9")}>
          9
        </div>
        <div className="btn orange" onClick={() => handleClick("X")}>
          X
        </div>
        <div className="btn" onClick={() => handleClick("4")}>
          4
        </div>
        <div className="btn" onClick={() => handleClick("5")}>
          5
        </div>
        <div className="btn" onClick={() => handleClick("6")}>
          6
        </div>
        <div className="btn orange" onClick={() => handleClick("+")}>
          +
        </div>
        <div className="btn" onClick={() => handleClick("1")}>
          1
        </div>
        <div className="btn" onClick={() => handleClick("2")}>
          2
        </div>
        <div className="btn" onClick={() => handleClick("3")}>
          3
        </div>
        <div className="btn orange" onClick={() => handleClick("-")}>
          -
        </div>
        <div className="btn" onClick={() => handleClick("0")}>
          0
        </div>
        <div className="btn" onClick={() => handleClick(".")}>
          .
        </div>
        <div className="btn" onClick={() => handleClick("=")}>
          =
        </div>
      </div>
    </div>
  );
}

export default Calculator;
