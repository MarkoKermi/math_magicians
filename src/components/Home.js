/* eslint-disable max-len */
// import design from '../design/Home.css';
// eslint-disable-next-line import/extensions
import design from './design/Home.css';

const Home = () => (
  <div className={design.homeContainer}>
    <h2 className={design.homeTitle}>Welcome to my Math Magicians website!</h2>
    <div className={design.homeParagraphs}>
      <p className={design.paragraphOne}>
        I would not dare to say that there is a direct relation between mathematics and madness, but there is no doubt that great mathematicians suffer from maniacal characteristics, delirium and symptoms of schizophrenia.
      </p>
      <p className={design.paragraphTwo}>
        Why are numbers beautiful? It’s like asking why is Beethoven’s Ninth Symphony beautiful. If you don’t see why, someone can’t tell you. I know numbers are beautiful. If they aren’t beautiful, nothing is.
      </p>
    </div>
  </div>
);
export default Home;
