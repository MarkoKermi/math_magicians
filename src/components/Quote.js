import design from './design/Quote.css';

const Quote = () => (
  <div className={design.quoteDiv}>
    <h2 className={design.quote}>
      Mathematics is not about numbers, equations,
      compulations or algorithms. It is about understanding.
      <p>--Wiliam Paul Thurston</p>
    </h2>
  </div>
);

export default Quote;
