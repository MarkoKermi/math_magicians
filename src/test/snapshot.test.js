import { render } from '@testing-library/react';
import Home from '../components/Home';
import Quote from '../components/Quote';
import Calculator from '../components/Calculator';

describe('Snapshot Test for Math Magicians project', () => {
  it('Snapshot needs to be equal as component render', () => {
    const viewHome = render(<Home />);
    expect(viewHome).toMatchSnapshot();
  });
  it('Snapshot needs to be equal as component render', () => {
    const viewQuote = render(<Quote />);
    expect(viewQuote).toMatchSnapshot();
  });
  it('Snapshot needs to be equal as component render', () => {
    const calculatorView = render(<Calculator />);
    expect(calculatorView).toMatchSnapshot();
  });
});
