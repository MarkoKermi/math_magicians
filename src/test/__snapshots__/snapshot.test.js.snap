// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Test for Math Magicians project Snapshot needs to be equal as component render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <h2>
          Welcome to my Math Magicians website!
        </h2>
        <div>
          <p>
            I would not dare to say that there is a direct relation between mathematics and madness; but there is no doubt that great mathematicians suffer from maniacal characteristics, delirium and symptoms of schizophrenia.
          </p>
          <p>
            Why are numbers beautiful? It is like asking why is Beethoven’s Ninth Symphony beautiful If you don’t see why, someone can’t tell you. I know numbers are beautiful. If they aren’t beautiful, nothing is.
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <h2>
        Welcome to my Math Magicians website!
      </h2>
      <div>
        <p>
          I would not dare to say that there is a direct relation between mathematics and madness; but there is no doubt that great mathematicians suffer from maniacal characteristics, delirium and symptoms of schizophrenia.
        </p>
        <p>
          Why are numbers beautiful? It is like asking why is Beethoven’s Ninth Symphony beautiful If you don’t see why, someone can’t tell you. I know numbers are beautiful. If they aren’t beautiful, nothing is.
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Snapshot Test for Math Magicians project Snapshot needs to be equal as component render 2`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <h2>
          Mathematics is not about numbers, equations, compulations or algorithms. It is about understanding.
          <p>
            --Wiliam Paul Thurston
          </p>
        </h2>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <h2>
        Mathematics is not about numbers, equations, compulations or algorithms. It is about understanding.
        <p>
          --Wiliam Paul Thurston
        </p>
      </h2>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Snapshot Test for Math Magicians project Snapshot needs to be equal as component render 3`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <div
          class="wrapper"
        >
          <input
            class="screen"
            type="text"
            value="0"
          />
          <button
            class="btn light-gray"
            type="button"
          >
            AC
          </button>
          <button
            class="btn light-gray"
            type="button"
          >
            %
          </button>
          <button
            class="btn light-gray"
            type="button"
          >
            +/-
          </button>
          <button
            class="btn orange"
            type="button"
          >
            ÷
          </button>
          <button
            class="btn"
            type="button"
          >
            7
          </button>
          <button
            class="btn"
            type="button"
          >
            8
          </button>
          <button
            class="btn"
            type="button"
          >
            9
          </button>
          <button
            class="btn orange"
            type="button"
          >
            x
          </button>
          <button
            class="btn"
            type="button"
          >
            4
          </button>
          <button
            class="btn"
            type="button"
          >
            5
          </button>
          <button
            class="btn"
            type="button"
          >
            6
          </button>
          <button
            class="btn orange"
            type="button"
          >
            +
          </button>
          <button
            class="btn"
            type="button"
          >
            1
          </button>
          <button
            class="btn"
            type="button"
          >
            2
          </button>
          <button
            class="btn"
            type="button"
          >
            3
          </button>
          <button
            class="btn orange"
            type="button"
          >
            -
          </button>
          <button
            class="btn zero"
            type="button"
          >
            0
          </button>
          <button
            class="btn"
            type="button"
          >
            .
          </button>
          <button
            class="btn"
            type="button"
          >
            =
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <div
        class="wrapper"
      >
        <input
          class="screen"
          type="text"
          value="0"
        />
        <button
          class="btn light-gray"
          type="button"
        >
          AC
        </button>
        <button
          class="btn light-gray"
          type="button"
        >
          %
        </button>
        <button
          class="btn light-gray"
          type="button"
        >
          +/-
        </button>
        <button
          class="btn orange"
          type="button"
        >
          ÷
        </button>
        <button
          class="btn"
          type="button"
        >
          7
        </button>
        <button
          class="btn"
          type="button"
        >
          8
        </button>
        <button
          class="btn"
          type="button"
        >
          9
        </button>
        <button
          class="btn orange"
          type="button"
        >
          x
        </button>
        <button
          class="btn"
          type="button"
        >
          4
        </button>
        <button
          class="btn"
          type="button"
        >
          5
        </button>
        <button
          class="btn"
          type="button"
        >
          6
        </button>
        <button
          class="btn orange"
          type="button"
        >
          +
        </button>
        <button
          class="btn"
          type="button"
        >
          1
        </button>
        <button
          class="btn"
          type="button"
        >
          2
        </button>
        <button
          class="btn"
          type="button"
        >
          3
        </button>
        <button
          class="btn orange"
          type="button"
        >
          -
        </button>
        <button
          class="btn zero"
          type="button"
        >
          0
        </button>
        <button
          class="btn"
          type="button"
        >
          .
        </button>
        <button
          class="btn"
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
